src/world/Units/Unit.Legacy.cpp

```diff
@@ -9066,15 +9058,12 @@ void Unit::AddAura(Aura* aur)
     }
 
     /* Set aurastates */
-    uint32 flag = 0;
     if (aur->GetSpellInfo()->getMechanicsType() == MECHANIC_ENRAGED && !asc_enraged++)
-        flag |= AURASTATE_FLAG_ENRAGED;
+        addAuraState(AURASTATE_FLAG_ENRAGED);
     else if (aur->GetSpellInfo()->getMechanicsType() == MECHANIC_BLEEDING && !asc_bleed++)
-        flag |= AURASTATE_FLAG_BLEED;
+        addAuraState(AURASTATE_FLAG_BLEED);
     if (aur->GetSpellInfo()->custom_BGR_one_buff_on_target & SPELL_TYPE_SEAL && !asc_seal++)
-        flag |= AURASTATE_FLAG_JUDGEMENT;
-
-    SetFlag(UNIT_FIELD_AURASTATE, flag);
```

Comment:
Why have you removed this call?

Reply:
because addAuraState adds the flag directly.

Reply:
Same as https://github.com/AscEmu/AscEmu/pull/572/files#r174589664